src/Workspaces/Core/Portable/Workspace/Host/SourceFiles/DynamicFileInfo.cs

```diff
@@ -37,6 +39,20 @@ public DynamicFileInfo(string filePath, SourceCodeKind sourceCodeKind, TextLoade
         /// </summary>
         public TextLoader TextLoader { get; }
 
+        /// <summary>
+        /// True if the source code contained in the document is only used in design-time (e.g. for completion),
+        /// but is not passed to the compiler when the containing project is built.
```

Comment:
these exist?  what's an example?  can we doc that example?

Reply:
Razor views is one example, at least.

Reply:
XAML also - they generate .g files that are design-time-only.

---
In reply to: [468926267](https://github.com/dotnet/roslyn/pull/46703#discussion_r468926267) [](ancestors = 468926267)

Reply:
Does XAML set this though?

Reply:
Ah, maybe not them.